src/operator/nn/softmax-inl.h

```diff
@@ -218,6 +219,157 @@ __global__ void softmax_compute_kernel(DType *in, OType *out, index_t M, int axi
   }
 }
 
+const int softmax_threads_per_block = 512;
+
+template <typename OP, typename T>
+__device__ inline T warp_reduce(T value, OP redfun) {
```

Comment:
this looks like a generic function that can be used elsewhere. Is there a better place to put this function?

Reply:
Yup, will look into putting it in some better place.

Reply:
It looks like common cuda utilities are put here https://github.com/apache/incubator-mxnet/blob/master/src/common/cuda_utils.h

Reply:
In fact, there are other common functions that spread in other files, e.g., the warp-level wrapper in https://github.com/apache/incubator-mxnet/blob/master/src/operator/nn/layer_norm.cu#L32-L50

Reply:
We can also put it into https://github.com/apache/incubator-mxnet/blob/master/src/operator/tensor/broadcast_reduce-inl.cuh